### PR TITLE
Edited README

### PR DESCRIPTION
**Pull request by kailash-manasarovar**
-> I'm closing the old branch as it became a merge conflict chaos.
-> I've tested most of what's in here now and can confirm that _most_ of the commands are working.
-> Set up GOOD.
-> Interact with blockchain GOOD.
-> Compile/deploy GOOD.
-> Bridge assets, deposit error, withdraw GOOD.

Then all commands related to deployed contracts I struggled with. I suggest someone run through these and test, ideally on local node to follow the actual instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,140 +194,52 @@ totalDifficulty      0
...
       --is-system    Enable the system contract compilation mode.
       --force-evmla  Sets the EVM legacy assembly pipeline forcibly
       -h, --help         Print help
-
-  
 ```
 
-### Compiling with `--is-system` flag
-It is necessary for some contracts, including those that deploy other contracts (such as factory contracts), to be compiled using the `--is-sytem` flag. These contracts should be placed in the `src/is-system/` folder, if the folder does not exist, manually create it:
+> `--is-system` flag: It is necessary to compile some contracts, including those that deploy other contracts (such as factory contracts), using the `--is-sytem` flag. These contracts should be placed in the `src/is-system/` folder. If the folder does not exist, manually create it.
```
Comment (githubdoramon) [inline README.md:219]: typo on the second --is-sytem (missing "s")

**Review by githubdoramon — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,132 +1,177 @@
-# ZkSync Smart Contract Testing, Deployment and Interaction Framework with Foundry
-### About
-Currently the industry standard Solidity smart contract test and deploy frameworks are Hardhat and Truffle. They both use JavaScript to test and deploy solidity smart contracts. Another smart contract testing platform by the name of Foundry. The  advantage that Foundry provides is that all tests are also written solidity creating a smoother developer experience. With Foundry, the engineer does not need to switch languages to write tests and deploy contracts.. 
+# Foundry with zkSync Era v0.0
```
[the remark this answers is not in the document]
Comment (kailash-manasarovar) [reply README.md:1]: could be, not sure

File: README.md (hunk 1 above)
Comment (kailash-manasarovar) [reply README.md:219]: there were a few of these, updated.

File: README.md [hunk 3]
```diff
@@ -149,140 +194,52 @@ totalDifficulty      0
...
-../foundry-zksync/target/debug/zkcast zk-send --withdraw 0x36615Cf349d7F6344891B1e7CA7C72883F5dc049 --amount 1000000 --rpc-url http://localhost:3050 --private-key 7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 --chain 270
-```
-#### Output
-```js
-Bridging assets....
-Transaction Hash: 0x94ef9e2eed345dcfef6f0b4f953f431b8edc6760e29b598a7cf446dab18d6317
-```
-
----
+Compile smart contracts to zkEvm bytecode and store the compiled output files in a logical directory structure `<PROJECT-ROOT>/zkout/` for easy retrieval by other components of the application.
```
[the remark this answers is not in the document]
Comment (kailash-manasarovar) [reply README.md]: yes looks like it, amended

File: README.md [hunk 4]
```diff
@@ -1,132 +1,177 @@
...
 
-Currently only Hardhat has developed functionality for zkSync. The purpose of this repository is to create functionality with Foundry to fully test, compile and deploy smart contracts on zkSync using only Solidity, as well as interact with those contracts once deployed.
+This repository provides [Foundry](https://github.com/foundry-rs/foundry) functionality in Solidity for compiling, deploying, testing, and interacting with smart contracts on zkSync Era.
+
+### Supported features
+
+- Compile smart contracts with the [zksolc compiler](https://github.com/matter-labs/zksolc-bin).
+- Deploy smart contracts to zkSync Era mainnet, testnet, or local test node.
+- Bridge assets L1 <-> L2.
+- Call deployed contracts on zkSync Era testnet or local test node.
```
[the remark this answers is not in the document]
Comment (kailash-manasarovar) [reply README.md:10]: I understand mainnet is OK.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mm-zk) [reply README.md:1]: yeah - let's update it to v0.1

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mm-zk) [reply README.md:10]: mainnet works too.

File: README.md [hunk 5]
```diff
@@ -1,132 +1,177 @@
...
-
-## L1_RPC_URL and L2_RPC_URL are used only for the zk-deposit command.
+## L1_RPC_URL and L2_RPC_URL are only used for `zk-deposit`
 L1_RPC_URL=https://localhost:8545
-# L2_RPC_URL can be used to replace --l2-url in command line for zk-deposit
+# L2_RPC_URL can be used to replace --l2-url in command line for `zk-deposit`
 L2_RPC_URL=https://localhost:3050
 
-# CHAIN can be used to replace --chain in command line  
+# CHAIN replaces --chain in command line
```
Comment (mm-zk) [inline README.md:79]: and 324 for mainnet

File: README.md [hunk 6]
```diff
@@ -149,140 +194,52 @@ totalDifficulty      0
...
 
-![image](https://user-images.githubusercontent.com/76663878/236301037-2a536ab0-3d09-44f3-a74d-5f5891af335b.png)
+        ![image](https://user-images.githubusercontent.com/76663878/236301037-2a536ab0-3d09-44f3-a74d-5f5891af335b.png)
 
-#### Example Usage
-To compile with only default compiler options (v1.3.9):
-```bash
+### Example usage
+
+To compile with default compiler options (v1.3.9).
```
Comment (mm-zk) [inline README.md:225]: default compiler is now 1.3.11

File: README.md [hunk 7]
```diff
@@ -331,33 +285,128 @@ ZkSync Features:
...
-## `zkcast zk-send`
-### aliases: `zkcast zks`, `zkcast zksend`
-Interact with deployed contracts in the native foundry/zkforge fashion using the CLI `zkcast zk-send` command:
+
+## Bridge assets L1 ↔ L2 with `zkcast zk-send` and `zkcast zk-deposit`
+
+### L1 → L2 deposits
+
+```sh
+zkcast zk-deposit <TO> <AMOUNT> <TOKEN> --rpc-url <RPC-URL> --l2-url <L2URL> --chain <CHAIN-ID> --private-key <PRIVATE-KEY>
```
Comment (mm-zk) [inline README.md:316]: zk deposit should be using the --l1-rpc-url flag (not --rpc-url)

File: README.md [hunk 8]
```diff
@@ -331,33 +285,128 @@ ZkSync Features:
...
+```
+NOTE: Leave `<TOKEN>` blank to bridge ETH
+
 ```bash
+Usage: zkcast zk-deposit  <TO> <AMOUNT> --rpc-url <ETH_RPC_URL> --l2-url <L2URL> [OPTIONS] [BRIDGE] [TIP]
+
+Arguments:
+  <TO>
+          The destination of the transaction.
```
Comment (mm-zk) [inline README.md:325]: this is the address on the L2, that will receive the tokens

File: README.md [hunk 9]
```diff
@@ -1,132 +1,177 @@
-# ZkSync Smart Contract Testing, Deployment and Interaction Framework with Foundry
-### About
-Currently the industry standard Solidity smart contract test and deploy frameworks are Hardhat and Truffle. They both use JavaScript to test and deploy solidity smart contracts. Another smart contract testing platform by the name of Foundry. The  advantage that Foundry provides is that all tests are also written solidity creating a smoother developer experience. With Foundry, the engineer does not need to switch languages to write tests and deploy contracts.. 
+# Foundry with zkSync Era v0.0
 
-Currently only Hardhat has developed functionality for zkSync. The purpose of this repository is to create functionality with Foundry to fully test, compile and deploy smart contracts on zkSync using only Solidity, as well as interact with those contracts once deployed.
+This repository provides [Foundry](https://github.com/foundry-rs/foundry) functionality in Solidity for compiling, deploying, testing, and interacting with smart contracts on zkSync Era.
```
[the remark this answers is not in the document]
Comment (kailash-manasarovar) [reply README.md:3]: removed

File: README.md (hunk 5 above)
Comment (kailash-manasarovar) [reply README.md:79]: added

File: README.md (hunk 6 above)
Comment (kailash-manasarovar) [reply README.md:225]: amended

File: README.md (hunk 7 above)
Comment (kailash-manasarovar) [reply README.md:316]: amended, this needs checking

File: README.md (hunk 8 above)
Comment (kailash-manasarovar) [reply README.md:325]: amended

**Pull request merged**